datafusion/src/physical_plan/hash_aggregate.rs

```diff
@@ -201,7 +203,9 @@ impl ExecutionPlan for HashAggregateExec {
 
     fn required_child_distribution(&self) -> Distribution {
         match &self.mode {
-            AggregateMode::Partial => Distribution::UnspecifiedDistribution,
+            AggregateMode::Partial | AggregateMode::FinalPartitioned => {
+                Distribution::UnspecifiedDistribution
```

Comment:
It seems like the required  required distribution for `FinalPartitioned` is really "hash partitioned" or some other partitioning that guarantees the same group key doesn't appear in multiple partitions. Leaving it as `UnspecifiedDistribution` might cause subtle bugs in the future.

Reply:
That's interesting - we could I think add a `Distribution::HashPartitioned(Vec<Expr>)` variant, which also could be used later for optimizations.
Currently only `SinglePartition` partition is used for the `MergeExec` rule.